### PR TITLE
Fix accuracy regression in Gemma

### PR DESCRIPTION
**Pull request by skavulya**
# What does this PR do?
Fixes accuracy regression in gemma from https://github.com/huggingface/optimum-habana/pull/1463  

Updates gemma-7b accuracy test with improved output.

`GAUDI2_CI=1  RUN_SLOW=1 python3.10 -m pytest tests/test_text_generation_example.py -s -vv -k google/gemma-7b-1-Fal --token $HF_TOKEN`

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by jiminha — APPROVED** [review]

File: optimum/habana/transformers/models/gemma/modeling_gemma.py [hunk 1]
```diff
@@ -652,7 +652,7 @@ def forward(
             position_ids = position_ids.unsqueeze(0)
 
         # HPU specific mask generation
-        if attention_mask is not None and attention_mask.dim() != 4:
+        if attention_mask is None or attention_mask.dim() != 4:
             attention_mask = _gaudi_prepare_4d_causal_attention_mask(
```
Comment (libinta) [inline optimum/habana/transformers/models/gemma/modeling_gemma.py:656]: if attention_mask is none, why do we still use this?

Comment (skavulya) [reply optimum/habana/transformers/models/gemma/modeling_gemma.py:656]: It creates a causal mask when attention_mask is None. The updated criteria is equivalent to https://github.com/huggingface/transformers/blob/53fad641cfdb5105e2470bcf3ef17ea8e25cc300/src/transformers/models/gemma/modeling_gemma.py#L532 

```
if attention_mask is not None and attention_mask.dim() == 4:
        # In this case we assume that the mask comes already in inverted form and requires no inversion or slicing.
        causal_mask = attention_mask
```

**Comment by skavulya** [comment]
Here are the results of  tranformer tests
 `GAUDI2_CI=1 RUN_SLOW=true python -m pytest tests/transformers/tests/models/`
Main commit 125e3d6b7049: ==== 28 failed, 986 passed, 421 skipped, 101 warnings in 1890.77s (0:31:30) ====
This PR: ==== 28 failed, 986 passed, 421 skipped, 101 warnings in 1538.51s (0:25:38) ====

**Comment by kaixuanliu** [comment]
Hi @skavulya , LGTM

**Review by regisss — APPROVED** [review]
